### PR TITLE
Change parameters order in Get-Help and help in order to get first -Full and then -Functionality when using Get-Help -Fu<tab> and help -Fu<tab>.

### LINKED ISSUE #7371: Change the order that -Full and -Functionality parameters of the Get-Help cmdlet show up when using Tab completion.
Since the **-Full** parameter of the `Get-Help` cmdlet (and `help` function) is used much more frequently than the **-Functionality** parameter it would be useful to change the order they show up when using Tab completion.

Also, alpabetically, **Full** goes earlier than **Functionality**.

Steps to reproduce
------------------

```powershell
Get-Help -Fu<tab>
```

Expected behavior
-----------------

```none
Get-Help -Full
```

Actual behavior
---------------

```none
Get-Help -Functionality
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.4
PSEdition                      Core
GitCommitId                    6.1.0-preview.4
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by sethvs**
Change parameters order in `Get-Help` cmdlet and `help` function in order to get first **-Full** and then **-Functionality** when using `Get-Help -Fu<tab>` and `help -Fu<tab>`.

Since **-Full** parameter is used much more frequently, than **-Functionality** it make sense to put it first when using tab completion.

Also, alphabetically, **Full** goes earlier than **Functionality**, that adds to the justification of the change.

Fix #7371 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 1]
```diff
@@ -180,6 +159,27 @@ public SwitchParameter Examples
         [Parameter(ParameterSetName = "Parameters", Mandatory = true)]
         public string Parameter { set; get; }
 
+        /// <summary>
+        /// List of Component's to search on.
+        /// </summary>
+        /// <value></value>
+        [Parameter]
+        public string[] Component { get; set; } = null;
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs]: Please remove initialization - null is default. Below too.

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 2]
```diff
@@ -180,6 +159,27 @@ public SwitchParameter Examples
         [Parameter(ParameterSetName = "Parameters", Mandatory = true)]
         public string Parameter { set; get; }
 
+        /// <summary>
+        /// List of Component's to search on.
+        /// </summary>
+        /// <value></value>
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs]: Please remove line `/// <value></value>`. Below too.

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 3]
```diff
@@ -180,6 +159,27 @@ public SwitchParameter Examples
         [Parameter(ParameterSetName = "Parameters", Mandatory = true)]
         public string Parameter { set; get; }
 
+        /// <summary>
+        /// List of Component's to search on.
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs]: Public comment for property should start with "Gets and sets ...". Below too.

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 4]
```diff
@@ -128,6 +128,18 @@ Describe "TabCompletion" -Tags CI {
         $res.CompletionMatches[0].CompletionText | Should -Be 'namespace'
     }
 
+    It 'Should first suggest -Full and then -Functionalty when using Get-Help -Fu<tab>' -skip {
+        $res = TabExpansion2 -inputScript 'Get-Help -Fu' -cursorColumn 'Get-Help -Fu'.Length
+        $res.CompletionMatches[0].CompletionText | Should -Be '-Full'
```
Comment (iSazonov) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Please use `-BeExactly` for strings. Below too.

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 5]
```diff
@@ -128,6 +128,18 @@ Describe "TabCompletion" -Tags CI {
...
+        $res.CompletionMatches[1].CompletionText | Should -Be '-Functionality'
+    }
+
+    It 'Should first suggest -Full and then -Functionalty when using help -Fu<tab>' -skip {
```
Comment (mklement0) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Typo (missing `i`): `Functionalty` -> `Functionality`

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 6]
```diff
@@ -128,6 +128,18 @@ Describe "TabCompletion" -Tags CI {
         $res.CompletionMatches[0].CompletionText | Should -Be 'namespace'
     }
 
+    It 'Should first suggest -Full and then -Functionalty when using Get-Help -Fu<tab>' -skip {
```
Comment (mklement0) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Typo (missing `i`): `Functionalty` -> `Functionality`

**Comment by mklement0** [comment]
Please see https://github.com/PowerShell/PowerShell/issues/7371#issuecomment-408233964 for why I think this problem should be tackled fundamentally differently.

File: src/System.Management.Automation/help/HelpCommands.cs (hunk 1 above)
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: Done.

File: src/System.Management.Automation/help/HelpCommands.cs (hunk 2 above)
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: Done.

File: src/System.Management.Automation/help/HelpCommands.cs (hunk 3 above)
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: Done.

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 (hunk 5 above)
Comment (sethvs) [reply test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Done.

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 (hunk 6 above)
Comment (sethvs) [reply test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Done.

**Review by iSazonov — APPROVED** [review]
I do not know when we will have a more general solution and so far this is LGTM.

**Comment by mklement0** [comment]
@iSazonov: Understood, but my point was that the effort expended here could have gone toward fixing the problem _fundamentally_.

**Comment by iSazonov** [comment]
@SteveL-MSFT Do you agree to merge the intermediate change?

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
I'm fine taking this change as-is.  @mklement0 Perhaps you can open a new issue to discuss the broader issue?

**Pull request merged**